### PR TITLE
NOD PDF generation incorrectly add extra page and incorrectly checks the extra issues checkbox

### PR DESCRIPTION
**Pull request by michelpmcdonald**
## Summary

Team banana peels code.

Found two bugs:

An extra empty "overflow" pages were being generated in the NOD PDF we send to CentralMail\EMMS API to establish NODs.  The check for a long rep name to trigger an extra page was incorrect, so even if there was not a long rep name that would not fit on the form, an extra blank page was still being generated

On the NOD form itself, the "extra issues on additional pages checkbox" was incorrectly checked, any extra pages causes the box to be checked, even if the extra pages were not due to more than 5 issues overflowing.

## Related issue(s)

[API-34725](https://jira.devops.va.gov/browse/API-34725)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots
Example of bug, Only two issues, no issues overflow page, but the issues overflow page checkbox is checked

image

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/form_data.rb [hunk 1]
```diff
@@ -79,6 +79,10 @@ def appealing_vha_denial
           appealing_vha_denial? ? 1 : 'Off'
         end
 
+        def extra_contestable_issues
+          contestable_issues.size > 5 ? 1 : 'Off'
+        end
+
```
Comment (michelpmcdonald) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/form_data.rb:85]: Added this specifically for the setting the extra issues checkbox on the form

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/pages/long_data_and_extra_issues.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def build!
           attr_accessor :pdf, :form_data
 
           def no_content
-            !extra_issues? && !form_data.long_preferred_email? && form_data.long_rep_name?
+            !extra_issues? && !form_data.long_preferred_email? && !form_data.long_rep_name?
```
Comment (michelpmcdonald) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/pages/long_data_and_extra_issues.rb:37]: hard to spot, but form_data.long_rep_name? was missing the not operator.  This was causing no_content to return false(ie there IS extra content) when in fact there was no extra content and no need for an extra overflow page

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb [hunk 3]
```diff
@@ -27,7 +29,7 @@ def form_fill
             form_fields.virtual_tele_hearing => form_data.virtual_tele_hearing,
             form_fields.requesting_extension => form_data.requesting_extension,
             form_fields.appealing_vha_denial => form_data.appealing_vha_denial,
-            form_fields.additional_issues => form_data.additional_pages,
+            form_fields.additional_issues => form_data.extra_contestable_issues,
```
Comment (michelpmcdonald) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb:32]: THis is the bug where we were checking the extra issues checkbox for any overflow pages, even if the overflow page are for long emails or long rep names.

Changed it to check the box only if there are more than 5 issues and an issues overflow page is added

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb [hunk 4]
```diff
@@ -148,20 +150,19 @@ def fill_first_five_issue_dates!(options)
           # to handle the contestableIssue content, so we fill the date, and do
           # the content afterwards.
 
-          form_data.contestable_issues.take(5).each_with_index do |issue, index|
+          form_data.contestable_issues.take(MAX_ISSUES_ON_MAIN_FORM).each_with_index do |issue, index|
```
Comment (michelpmcdonald) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb:153]: use the pre existing constant

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb [hunk 5]
```diff
@@ -148,20 +150,19 @@ def fill_first_five_issue_dates!(options)
...
             options[form_fields.issue_table_decision_date(index)] = issue['attributes']['decisionDate']
           end
 
           options
         end
-        # rubocop:disable Layout/LineLength
 
         def additional_pages?
-          form_data.contestable_issues.count > 5 || form_data.long_preferred_email? || form_data.requesting_extension? || form_data.long_rep_name?
+          form_data.contestable_issues.count > MAX_ISSUES_ON_MAIN_FORM ||
```
Comment (michelpmcdonald) [inline modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/structure.rb:161]: use the pre-existing constant

File: modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/pages/long_data_and_extra_issues.rb (hunk 2 above)
Comment (kristen-brown) [reply modules/appeals_api/app/services/appeals_api/pdf_construction/notice_of_disagreement/v2/pages/long_data_and_extra_issues.rb:37]: Nice catch!

**Review by kristen-brown — APPROVED** [review]
Looks good to me!

**Pull request merged**